nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-ssl-autoloading-utils/src/test/java/org/apache/nifi/autoload/TestSSLContextFactoryAutoLoaderTask.java

```diff
@@ -0,0 +1,113 @@
+    private NiFiProperties nifiProperties;
+
+    @Before
+    public void init() throws GeneralSecurityException, IOException {
+        this.createKeystore(DEFAULT_KEY_ALIAS, DEFAULT_CERT_DN);
+
+        nifiProperties = Mockito.mock(NiFiProperties.class);
+        Mockito.when(nifiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_PASSWD)).thenReturn(KEYSTORE_PASSWORD);
+        Mockito.when(nifiProperties.getProperty(NiFiProperties.SECURITY_KEY_PASSWD)).thenReturn(KEY_PASSWORD);
+        Mockito.when(nifiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_TYPE)).thenReturn(KEYSTORE_TYPE);
+
+        task = new SSLContextFactoryAutoLoaderTask.Builder()
+                .keystorePath(KEYSTORE)
+                .keystoreWatchService(Mockito.mock(WatchService.class))
+                .truststorePath(Mockito.mock(Path.class))
+                .autoLoader(Mockito.mock(SSLContextFactoryAutoLoader.class))
+                .nifiProperties(nifiProperties)
+                .pollIntervalMillis(100)
+                .build();
+    }
+
+    @Test
+    public void testIsReloadAllowed_true() throws IOException, GeneralSecurityException {
+        this.createKeystore(DEFAULT_KEY_ALIAS, DEFAULT_CERT_DN); // Creates a keystore with the same DNs and alias, so this is allowed
+        Assert.assertTrue(task.isReloadAllowed());
+    }
+
+    @Test
+    public void testIsReloadAllowed_differentAlias() throws IOException, GeneralSecurityException {
+        this.createKeystore("different-alias", DEFAULT_CERT_DN);
+        Assert.assertFalse(task.isReloadAllowed());
+    }
+
+    @Test
+    public void testIsReloadAllowed_differentSubjectDN() throws IOException, GeneralSecurityException {
+        this.createKeystore(DEFAULT_KEY_ALIAS, "CN=different");
+        Assert.assertFalse(task.isReloadAllowed());
+    }
+
+    @Test
+    public void testIsReloadAllowed_differentIssuerSerialNumber() throws IOException, UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException, TlsException {
+        // This is a keystore with the same key alias and cert DN, but with a different issuer cert in the cert chain,
+        // showing that we disallow cert updates with the same issuer DN but different actual issuer serial number
+        Files.copy(new File("src/test/resources/test-keystore.jks").toPath(), KEYSTORE, StandardCopyOption.REPLACE_EXISTING);
+
+        Assert.assertFalse(task.isReloadAllowed());
+    }
+
+    private void createKeystore(String keyAlias, String certDn) throws IOException, GeneralSecurityException {
+        TlsConfiguration tlsConfiguration = KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore(new StandardTlsConfiguration(
+                null, KEYSTORE_PASSWORD, KEY_PASSWORD, KEYSTORE_TYPE,
+                null, KEYSTORE_PASSWORD, KEYSTORE_TYPE), keyAlias, "nifi-cert", certDn);
+        Files.move(new File(tlsConfiguration.getKeystorePath()).toPath(), KEYSTORE, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
According to the automated build, Windows does not support the `StandardCopyOption.ATOMIC_MOVE`, so it looks like replacing this with `StandardCopyOption.REPLACE_EXISTING` should work.

Reply:
Nice, this also helped me find ATOMIC_MOVE in ConfigEncryptionTool.